### PR TITLE
Use plugin-installation-manager-tool

### PR DESCRIPTION
**Pull request by halkeye**
# What this PR does / why we need it

Updates the helm chart to use the plugin-installation-manager-tool

# Special notes for your reviewer

I've created a PR to add the new version to the docker images, so it'll soon be working.
Note: before 2.1.1 it tries to delete the entire directory where plugins are installed into, which breaks in the helm chart because its a volume - Fixed in https://github.com/jenkinsci/plugin-installation-manager-tool/pull/215

Added a version check and added https://github.com/jenkinsci/docker/pull/1015 so it gets released in new docker images

I don't know how to run the tests though

# Checklist

- [X] [DCO](https://github.com/jenkinsci/helm-charts/blob/main/CONTRIBUTING.md#sign-off-your-work) signed
- [X] Chart Version bumped
- [X] CHANGELOG.md was updated

### CODE REVIEW

**Comment by halkeye** [comment]
I tested it locally by doing:
```
diff --git a/charts/jenkins/templates/config.yaml b/charts/jenkins/templates/config.yaml
index 9cb94c6..616a9cf 100644
--- a/charts/jenkins/templates/config.yaml
+++ b/charts/jenkins/templates/config.yaml
@@ -158,6 +158,7 @@ data:
     </jenkins.CLI>
 {{- end }}
   apply_config.sh: |-
+    set -x
 {{- if .Values.master.initializeOnce }}
     if [ -f {{ .Values.master.jenkinsHome }}/initialization-completed ]; then
       echo "master was previously initialized, refusing to re-initialize"
@@ -211,6 +212,10 @@ data:
     rm -rf {{ .Values.master.jenkinsRef }}/plugins/*.lock
     version () { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }

+    ## TEMP START
+    alias jenkins-plugin-cli="java -jar /tmp/jenkins-plugin-manager.jar"
+    wget -o /dev/null -O /tmp/jenkins-plugin-manager.jar https://github.com/jenkinsci/plugin-installation-manager-tool/releases/download/2.1.1/jenkins-plugin-manager-2.1.1.jar && chmod 777 /tmp/jenkins-plugin-manager.jar;
+    ## TEMP END

     if [ -n "$(command -v jenkins-plugin-cli)" 2>/dev/null ] && [ $(version $(jenkins-plugin-cli --version)) -ge $(version "2.1.1") ]; then
       jenkins-plugin-cli -w /usr/share/jenkins/jenkins.war --verbose -f {{ .Values.master.jenkinsHome }}/plugins.txt;

```

then 🎉 

image

**Comment by torstenwalter** [comment]
Wow! This looks great!

**Comment by torstenwalter** [comment]
I think the test failed because the checksum of the config map changed.

**Comment by halkeye** [comment]
Oh havnt even gotten to that error yet. I gotta update it to match the changes but I keep making tiny mistakes

**Comment by halkeye** [comment]
ill do another pass soon

File: charts/jenkins/templates/config.yaml [hunk 1]
```diff
@@ -209,7 +209,14 @@ data:
     # Install missing plugins
     cp /var/jenkins_config/plugins.txt {{ .Values.master.jenkinsHome }};
     rm -rf {{ .Values.master.jenkinsRef }}/plugins/*.lock
-    /usr/local/bin/install-plugins.sh `echo $(cat {{ .Values.master.jenkinsHome }}/plugins.txt)`;
+    version () { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+
+
+    if [ -n "$(command -v jenkins-plugin-cli)" 2>/dev/null ] && [ $(version $(jenkins-plugin-cli --version)) -ge $(version "2.1.1") ]; then
+      jenkins-plugin-cli -w /usr/share/jenkins/jenkins.war --verbose -f {{ .Values.master.jenkinsHome }}/plugins.txt;
```
Comment (timja) [inline charts/jenkins/templates/config.yaml:216]: Do you need verbose on?

Comment (timja) [inline charts/jenkins/templates/config.yaml:216]: Can we use the long argument —war instead? I think this is the default so it shouldn’t actually be needed too

Comment (halkeye) [reply charts/jenkins/templates/config.yaml:216]: no, I added it when trying to figure out what was going on, but never turned it off. I'll make sure its removed next commmit

Comment (halkeye) [reply charts/jenkins/templates/config.yaml:216]: > Can we use the long argument —war instead? 

Done

> I think this is the default so it shouldn’t actually be needed too

Agreed, except the ref directory is configurable, which i missed. so I fix.

**Comment by halkeye** [comment]
yay all checks pass

File: charts/jenkins/tests/config-test.yaml [hunk 2]
```diff
@@ -27,7 +27,12 @@ tests:
             # Install missing plugins
             cp /var/jenkins_config/plugins.txt /var/jenkins_home;
             rm -rf /usr/share/jenkins/ref/plugins/*.lock
-            /usr/local/bin/install-plugins.sh `echo $(cat /var/jenkins_home/plugins.txt)`;
+            version () { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+            if [ -n "$(command -v jenkins-plugin-cli)" 2>/dev/null ] && [ $(version $(jenkins-plugin-cli --version)) -ge $(version "2.1.1") ]; then
+              jenkins-plugin-cli --war /usr/share/jenkins/ref/jenkins.war --plugin-file /var/jenkins_home/plugins.txt;
```
Comment (torstenwalter) [inline charts/jenkins/tests/config-test.yaml:32]: Do we need to check if `/usr/share/jenkins/ref/jenkins.war` exists?

File: charts/jenkins/templates/config.yaml [hunk 3]
```diff
@@ -209,7 +209,12 @@ data:
     # Install missing plugins
     cp /var/jenkins_config/plugins.txt {{ .Values.master.jenkinsHome }};
     rm -rf {{ .Values.master.jenkinsRef }}/plugins/*.lock
-    /usr/local/bin/install-plugins.sh `echo $(cat {{ .Values.master.jenkinsHome }}/plugins.txt)`;
+    version () { echo "$@" | awk -F. '{ printf("%d%03d%03d%03d\n", $1,$2,$3,$4); }'; }
+    if [ -n "$(command -v jenkins-plugin-cli)" 2>/dev/null ] && [ $(version $(jenkins-plugin-cli --version)) -ge $(version "2.1.1") ]; then
+      jenkins-plugin-cli --war {{ .Values.master.jenkinsRef }}/jenkins.war --plugin-file {{ .Values.master.jenkinsHome }}/plugins.txt;
```
Comment (torstenwalter) [inline charts/jenkins/templates/config.yaml:214]: Do we need to check if `{{ .Values.master.jenkinsRef }}/jenkins.war` exists?

Comment (halkeye) [reply charts/jenkins/templates/config.yaml:214]: i don't know, do we? doesn't most of the chart assume ref has content?

Comment (torstenwalter) [reply charts/jenkins/templates/config.yaml:214]: Not sure myself. It's possible that people use their own image. Need to check other references to see if this is a problem.

Comment (halkeye) [reply charts/jenkins/templates/config.yaml:214]: I have the plugins installed directly into my image, so I personally disable the install plugins feature (I like being able to roll back and not need the internet to startup)

lemme know if you want me to add the if check

Comment (torstenwalter) [reply charts/jenkins/templates/config.yaml:214]: @halkeye can you ping me once you added the check?

Comment (halkeye) [reply charts/jenkins/templates/config.yaml:214]: @torstenwalter totally got distracted and didn't get to it.

Added a check to use the home dir war file if the ref doesn't exist

**Comment by halkeye** [comment]
oh man I hate that DCO plugin. I'll get it fixed

**Review by torstenwalter — APPROVED** [review]
@halkeye Thank you! This one is really nice!

**Review by timja — APPROVED** [review]

**Pull request merged**